xla/service/gpu/gpu_prim.h

```diff
@@ -44,10 +44,9 @@ __device__ __forceinline__ void ThreadStoreVolatilePtr<Eigen::half>(
       Eigen::numext::bit_cast<uint16_t>(val);
 }
 
-template <>
-__device__ __forceinline__ Eigen::half ThreadLoadVolatilePointer<Eigen::half>(
```

Comment:
What about a unit test ensuring this works as expected?

Reply:
I'm not sure how to write one. I'm replacing the template specialization with an overload, but the body of the function stays the same. I can't think of a test that behaves differently before and after my patch.

If you have an idea for a good test, let me know. Otherwise, IMO the PR is OK to merge as-is.

Reply:
Alright, this one is ok to submit.